src/components/Search/Search.tsx

```diff
@@ -1,20 +1,43 @@
-import React, { useEffect, useState } from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
 import './Search.css';
 
-export const Search = () => {
-  const [search, setSearch] = useState((localStorage.getItem('search') as string) || 'Http cats');
+type TSearch = {
+  onCards: Dispatch<SetStateAction<never[]>>;
+  error: Dispatch<SetStateAction<null>>;
+};
+
+export const Search = (props: TSearch) => {
+  const [search, setSearch] = useState((localStorage.getItem('search') as string) || '');
+  const { onCards } = props;
+  const [input, setInput] = useState('');
 
   const onValueChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
     setSearch(e.target.value);
   };
 
-  useEffect(() => {
+  const handleSubmit = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();

```

Comment:
no need pass event if you are not using it

Reply:
I answered you in the chat, where you checked my work, please look